packages/fast-components/src/slider/slider.ts

```diff
@@ -0,0 +1,327 @@
+
+    /**
+     * Min allowed value
+     */
+    @attr
+    public min: number = 0; // Map to proxy element.
+    private minChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.min = `${this.min}`;
+        }
+    }
+
+    /**
+     * Max allowed value
+     */
+    @attr
+    public max: number = 10; // Map to proxy element.
+    private maxChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.max = `${this.max}`;
+        }
+    }
+
+    @attr
+    public step: number = 1; // Map to proxy element.
+    private stepChanged(): void {
+        if (this.proxy instanceof HTMLElement) {
+            this.proxy.step = `${this.step}`;
+        }
+    }
+
+    /**
+     * Orientation value, horizontal or vertical
+     */
+    @attr
+    public orientation: SliderOrientation = SliderOrientation.horizontal;
+
+    /**
+     * mode value, singleValue | dualThumb | adjustFromUpper | adjustFromLower
+     */
+    @attr
+    public mode: SliderMode = SliderMode.singleValue;
+
+    /**
+     * Set to true when the component has constructed
+     */
+    private constructed: boolean = false;
+    protected proxy = document.createElement("input");
+
+    constructor() {
+        super();
+
+        this.proxy.setAttribute("type", "range");
+        this.setAttribute("role", "slider");
+        this.setAttribute("tabindex", "0");
+        this.direction = this.getDirection();
+        this.constructed = true;
+    }
+
+    public connectedCallback(): void {
+        super.connectedCallback();
+        this.updateForm();
+        this.setupTrackConstraints();
+        this.setupListeners();
+        this.setupClassForOrientation();
+        this.setupDefaultValue();
+    }
+
+    public disconnectedCallback(): void {
+        this.removeEventListener("keydown", this.keypressHandler);
+        this.removeEventListener("mousedown", this.clickHandler);
+    }
+
+    protected keypressHandler = (e: KeyboardEvent) => {
+        super.keypressHandler(e);
+        switch (e.keyCode) {
+            case keyCodeArrowRight:
+            case keyCodeArrowUp:
+                this.increment();
+                break;
+            case keyCodeArrowLeft:
+            case keyCodeArrowDown:
+                this.decrement();
+                break;
+        }
+    };
+
+    private getDirection = (): string => {
+        const dirNode: any = this.parentElement!.closest("div[dir]");
+        return dirNode !== null && dirNode["dir"] === "rtl" ? "rtl" : "ltr";
+    };
+
+    private setupTrackConstraints = (): void => {
+        this.trackWidth = this.track.clientWidth;
+        this.trackMinWidth = this.track.getBoundingClientRect().left;
+    };
+
+    private setupListeners = (): void => {
+        this.addEventListener("keydown", this.keypressHandler);
+        this.addEventListener("mousedown", this.clickHandler);
+        this.thumb.addEventListener("mousedown", this.handleThumbMouseDown);
```

Comment:
You don't remove the thumb listener in teardown but probably should

Reply:
done